client/app/modals/tournament-status/tournament-status.component.ts

```diff
@@ -0,0 +1,104 @@
+import { Component, OnInit } from '@angular/core';
+import { TournamentStatusService } from '../../services/tournament-status.service';
+import { forkJoin } from '../../../../node_modules/rxjs';
+
+@Component({
+  selector: 'tournament-status',
+  templateUrl: './tournament-status.component.html',
+  styleUrls: ['./tournament-status.component.css']
+})
+export class TournamentStatusComponent implements OnInit {
+  private _matchChangeTime
+  private _secondsUntilMatchOnChange
+  private _matchHasChanged = false
+  private _stringTimeUntilMatch = 'Waiting for data'
+  private _tournamentStatus
+  private _currentMatch
+  private _currentMatchNumber
+  private _lastMatchStatus = {'text': 'Last match!', 'color': '#FFFFFF'}
+  private _finishedAllMatchesStatus = {'text':'All done!', 'color': '#FFFFFF'}
+
+  constructor(private tournamentStatusService: TournamentStatusService) {
+   }
+
+  ngOnInit() {
+    this.updateVariables()
+    setInterval(() => {
+      this.matchChanged(this._currentMatch)
+      if(this._matchHasChanged){
+        if(this._currentMatchNumber !== 0){
+          this.tournamentStatusService.getCurrentMatch().subscribe(()=>{
+            this.updateVariables()
+          },
+          err =>{
+            this._secondsUntilMatchOnChange = null
+            this._stringTimeUntilMatch = ''
+            this._currentMatchNumber = null
+            this._tournamentStatus = this._finishedAllMatchesStatus
+          })
+        }else{
+          this.updateVariables()
+        }
+      }
+      this._matchHasChanged = false
+    }, 1000)
+  }
+
+  private matchChanged(match){
+    forkJoin(match, this.tournamentStatusService.getMatch()).subscribe(data =>{
+      this._matchHasChanged = data[0] !== data[1]
```

Comment:
this is not the right way for checking it the match has changed.
After discussing with Idan, he reminded me that one of our goals was the ability to handle weak networks.
You need to change it to reload the matches after getting an mhub message and doing each seconds only an update to the UI.